src/dal/taskDal.ts

```
@@ -2,7 +2,15 @@ import { tasks } from "../data/tasksStore";
 import { CreateTaskDetails, Task } from "../taskDto";
 import { v4 as uuidv4 } from "uuid";
 
-const getAllTasks = (): Task[] => {
+const getTasks = (filter: string | undefined): Task[] | undefined => {
+    if (filter) {
+        if (filter === 'myday') {
+            return tasks.filter((task) => task.myDay === true);
+        }
+
+        return undefined;
```

its better to throw error here since you reached an unexpected filter which breaks the whole logic